File: haushaltsbuch/src/test/java/org/github/haschi/infrastruktur/EventStoreLieferant.java
package org.github.haschi.infrastruktur;

import org.axonframework.config.Configuration;
import org.axonframework.eventsourcing.eventstore.EmbeddedEventStore;
import org.axonframework.eventsourcing.eventstore.EventStorageEngine;
import org.axonframework.eventsourcing.eventstore.EventStore;
import org.axonframework.eventsourcing.eventstore.inmemory.InMemoryEventStorageEngine;

@SuppressWarnings("WeakerAccess")
public final class EventStoreLieferant
{

    private final EventStorageEngine engine = new InMemoryEventStorageEngine();
    private final EmbeddedEventStore eventStore = new EmbeddedEventStore(engine);

    public EventStore eventBus(final Configuration konfiguration)
    {
        return eventStore;
    }
}

File: haushaltsbuch-domain/src/test/kotlin/org/github/haschi/infrastruktur/EventStoreLieferant.kt
package org.github.haschi.infrastruktur

import org.axonframework.config.Configuration
import org.axonframework.eventsourcing.eventstore.EmbeddedEventStore
import org.axonframework.eventsourcing.eventstore.EventStore
import org.axonframework.eventsourcing.eventstore.inmemory.InMemoryEventStorageEngine

class EventStoreLieferant
{

    private val engine = InMemoryEventStorageEngine()
    private val eventStore = EmbeddedEventStore(engine)

    fun eventBus(konfiguration: Configuration): EventStore
    {
        return eventStore
    }
}
